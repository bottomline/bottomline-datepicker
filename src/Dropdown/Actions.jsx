import injectSheet from 'react-jss';
import React from 'react';
import PropTypes from 'prop-types';
import localeGet from '../utils/locale.js';
import styles from './Actions.styles';

const Actions = ({
  classes,
  onClickApply,
  onClickCancel,
  disabled,
  dataQa,
  buttonClasses,
  applyButtonClasses,
  cancelButtonClasses
}) => (
  <div className={`${classes.actions} ${buttonClasses}`}>
    <button
      className={`${classes.applyButton}  ${applyButtonClasses}`}
      data-qa={dataQa ? `${dataQa}-apply-button` : 'apply-button'}
      disabled={disabled}
      onClick={onClickApply}
      type="button"
    >
      {localeGet('apply')}
    </button>
    <button
      type="button"
      data-qa={dataQa ? `${dataQa}-cancel-button` : 'cancel-button'}
      onClick={onClickCancel}
      className={`${classes.cancelButton} ${cancelButtonClasses}`}
    >
      {localeGet('cancel')}
    </button>
  </div>
);

Actions.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  onClickApply: PropTypes.func.isRequired,
  onClickCancel: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  dataQa: PropTypes.string,
  buttonClasses: PropTypes.string.isRequired,
  applyButtonClasses: PropTypes.string.isRequired,
  cancelButtonClasses: PropTypes.string.isRequired,
  options: PropTypes.shape({})
};

Actions.defaultProps = {
  disabled: false,
  dataQa: '',
  options: {}
};

export default injectSheet(styles)(Actions);
