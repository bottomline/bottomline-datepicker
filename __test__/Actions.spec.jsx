import React from 'react';
import { mount } from 'enzyme';
import Actions from '../src/Dropdown/Actions';

describe('Actions', () => {
  afterEach(() => {
    document.body.innerHTML = '';
  });

  it('should render correctly', () => {
    const wrapper = mount(
      <Actions {...{
        onClickApply: () => {},
        onClickCancel: () => {},
        disabled: false,
        dataQa: undefined,
        buttonClasses: '',
        applyButtonClasses: '',
        cancelButtonClasses: ''
      }}
      />
    );
    const button = wrapper.find('button');
    expect(button).toMatchSnapshot();
  });

  it('should render correctly with defined dataQa', () => {
    const wrapper = mount(
      <Actions {...{
        onClickApply: () => {},
        onClickCancel: () => {},
        disabled: false,
        dataQa: 'khorne',
        buttonClasses: '',
        applyButtonClasses: '',
        cancelButtonClasses: ''
      }}
      />
    );
    const button = wrapper.find('button');
    expect(button).toMatchSnapshot();
  });
});
