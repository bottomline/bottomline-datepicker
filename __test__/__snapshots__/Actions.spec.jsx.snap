// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions should render correctly 1`] = `
Array [
  <button
    className="Actions-applyButton-0-1-1  "
    data-qa="apply-button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Apply
  </button>,
  <button
    className="Actions-cancelButton-0-1-2 "
    data-qa="cancel-button"
    onClick={[Function]}
    type="button"
  >
    Cancel
  </button>,
]
`;

exports[`Actions should render correctly with defined dataQa 1`] = `
Array [
  <button
    className="Actions-applyButton-0-1-1  "
    data-qa="khorne-apply-button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Apply
  </button>,
  <button
    className="Actions-cancelButton-0-1-2 "
    data-qa="khorne-cancel-button"
    onClick={[Function]}
    type="button"
  >
    Cancel
  </button>,
]
`;
